assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -8234,6 +8422,41 @@ private void setNewAssignmentParameters(RunData data, boolean validify) {
 
     } // setNewAssignmentParameters
 
+    // What this part does is search in one gradebook or several (if isGradebookGroupEnabled is active)
+    // for the category that contains the assignment. When it finds it, it retrieves the points for the
+    // category (category.getPointsForCategory()) and stores them in a HashMap that keeps track
+    // of the gradebook + category score. In the end, instead of using the old Double,
+    // we will iterate through the HashMap, using the Double from each entry, and perform
+    // the same check as before.
+    private void buildGradebookPointsMap(String gbUid, String siteId, String assignmentRef,
```

Comment:
this should likely be something the gradingService should do, remember this feature is about enabling group gradebooks and then ensuring tools integrate with that.

Any time your having to get data from the GradingService and then filter or manipulate it **in a tool** then you should likely should consider moving the method to the GradingService and have it do that. Remember the tools are users of the GradingService.